### PR TITLE
Fix import order of archived data

### LINKED ISSUE #13498: Import configuration fails with foreign key constraint error
## Environment
- **Airbyte version**:  0.39.9-alpha
- **OS Version / Instance**: macOS (x86)
- **Deployment**: Docker
- **Source Connector and version**: source-snowflake:0.1.12
- **Destination Connector and version**: destination-s3:0.3.6
- **Step where error happened**: Import configuration

## Current Behavior
After creating a connection with snowflake as the source and s3 as the destination, I am able to export the configuration from the UI, but get a foreign key constraint error when trying to import the resulting configuration

## Expected Behavior
I should be able to import the configuration without an error

## Logs
Attached to this issue. look for: 
```
ERROR: insert or update on table "connection" violates foreign key constraint "connection_actor_catalog_id_fk"
  Detail: Key (source_catalog_id)=(1d033f52-d6a4-4256-98cc-34c607f2b209) is not present in table "actor_catalog".
```
[scheduler-logs.txt](https://github.com/airbytehq/airbyte/files/8845559/scheduler-logs.txt)
[server-logs.txt](https://github.com/airbytehq/airbyte/files/8845560/server-logs.txt)

## Steps to Reproduce
1. Create a snowflake source
2. Create an s3 destination
3. Create a connection linking the source to the destination
4. go to settings in the ui and export the configuration
5. go to settings in the ui and import the configuration produced by step 4
6. Observe that the import fails, and server logs output a foreign key constraint error

## Are you willing to submit a PR?
Possibly! It would take me some time to understand how the configuration import works and I may need some pointers on where to look for debugging

### PR DESCRIPTION
**Pull request by malikdiarra**

### CODE REVIEW

**Review by alafanechere — APPROVED** [review]
I confirm that I can now run an import 🙏 thank you!

File: airbyte-server/src/test/java/io/airbyte/server/handlers/ArchiveHandlerTest.java [hunk 1]
```diff
@@ -208,18 +211,47 @@ void testFullExportImportRoundTrip() throws Exception {
...
+    final SourceConnection source = new SourceConnection()
         .withSourceDefinitionId(sourceS3DefinitionId)
         .withSourceId(UUID.randomUUID())
-        .withWorkspaceId(standardWorkspace.getWorkspaceId())
+        .withWorkspaceId(workspace.getWorkspaceId())
         .withName("Test source")
         .withConfiguration(Jsons.deserialize("{}"))
         .withTombstone(false);
 
+    final StandardDestinationDefinition DESTINATION_S3 = new StandardDestinationDefinition()
```
Comment (davinchia) [inline airbyte-server/src/test/java/io/airbyte/server/handlers/ArchiveHandlerTest.java:222]: @alafanechere how did you test this works? I buffed up this test case to try and reproduce this. However, this is passing regardless of serialisation order in DatabaseConfigPersistence.java

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java [hunk 2]
```diff
@@ -1465,6 +1450,21 @@ public void replaceAllConfigs(final Map<AirbyteConfig, Stream<?>> configs, final
         LOGGER.warn(ConfigSchema.ACTOR_CATALOG_FETCH_EVENT + NOT_FOUND);
       }
 
+      if (configs.containsKey(ConfigSchema.STANDARD_SYNC)) {
```
[the remark this answers is not in the document]
Comment (davinchia) [reply airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java:1455]: add a comment to explain why

File: airbyte-server/src/test/java/io/airbyte/server/handlers/ArchiveHandlerTest.java (hunk 1 above)
Comment (davinchia) [reply airbyte-server/src/test/java/io/airbyte/server/handlers/ArchiveHandlerTest.java:224]: figure this out!

**Pull request merged**